easybuild/easyblocks/t/tensorflow.py

```diff
@@ -562,6 +562,15 @@ def configure_step(self):
         tensorrt_root = get_software_root('TensorRT')
         nccl_root = get_software_root('NCCL')
 
+        # add path to libnccl.so.2 directory provided by NCCL when both sysroot
+        # and RPATH are used (such as in EESSI)
+        if build_option('sysroot') and self.toolchain.use_rpath:
+            system_libs_info_as_list = list(self.system_libs_info)
+            new_libpaths = system_libs_info_as_list[2]
+            new_libpaths.append(os.path.join(nccl_root, 'lib'))
+            system_libs_info_as_list[2] = new_libpaths
+            self.system_libs_info = tuple(system_libs_info_as_list)
```

Comment:
```suggestion
            system_libs_info_as_list = list(self.system_libs_info)
            system_libs_info_as_list[2].append(get_software_libdir('NCCL'))
            self.system_libs_info = tuple(system_libs_info_as_list)
```

Reply:
Will test if that works incl moving the code to the `if` block for `nccl_root`.

Reply:
This almost worked. `get_software_libdir('NCCL')` returns the directory name, not the full path. Slightly modified this, see https://github.com/easybuilders/easybuild-easyblocks/pull/3497/commits/f6a9afd6fc8a9794d7a2879986db5f97acf9663d The latter version worked.